papcli/main.go

```diff
@@ -5,26 +5,57 @@ import (
 	"io/ioutil"
 	"path/filepath"
 
-	log "github.com/Sirupsen/logrus"
 	pdpctrl "github.com/infobloxopen/themis/pdpctrl-client"
+
+	log "github.com/Sirupsen/logrus"
 )
 
 func main() {
 	log.SetLevel(log.InfoLevel)
 
-	policy, includes, err := read(config.Policy, config.Includes)
+	policies, includes, err := read(config.Policy, config.Includes)
 	if err != nil {
 		panic(err)
 	}
 
-	cluster := pdpctrl.NewCluster(config.Addresses, config.Chunk, log.StandardLogger())
-	err = cluster.Connect(config.Timeout)
-	if err != nil {
-		panic(err)
+	hosts := []*pdpctrl.Client{}
+
+	for _, addr := range config.Addresses {
+		h := pdpctrl.NewClient(addr, config.Chunk)
+		if err := h.Connect(config.Timeout); err != nil {
+			panic(err)
+		}
+
+		hosts = append(hosts, h)
+		defer h.Close()
 	}
-	defer cluster.Close()
 
-	cluster.Process(includes, policy)
+	log.Infof("Uploading data to PDP servers...")
+
+	bids := make([]int32, len(hosts))
+	for i, h := range hosts {
+		b := &pdpctrl.DataBucket{
+			Policies: policies,
+			Includes: includes,
+		}
+
+		if err := h.Upload(b); err != nil {
+			log.Errorf("Failed to upload data: %v", err)
```

Comment:
This should be responsibility of client's user (e.g. PAP implementation) because we need to recreate binary content and upload new data bucket.

Also version error contains correct version number so we not necessary need entire copy